consensus/src/agreement/verifiers.rs

```diff
@@ -136,22 +139,30 @@ pub async fn verify_step_votes(
         &sv.aggregate_signature.inner(),
         committees_set,
         &cfg,
+        enable_quorum_check,
     )
     .await
 }
 
+#[allow(clippy::too_many_arguments)]
 pub async fn verify_votes(
     block_hash: &[u8; 32],
     bitset: u64,
     signature: &[u8; 48],
     committees_set: &Arc<Mutex<CommitteeSet>>,
     cfg: &sortition::Config,
+    enable_quorum_check: bool,
 ) -> Result<(), Error> {
-    let sub_committee = {
-        // Scoped guard to fetch committee data quickly
+    // TODO: This should be refactored into a structure when #1118 issue is
+    // implemented
+    let sub_committee = 'label: {
         let mut guard = committees_set.lock().await;
-
         let sub_committee = guard.intersect(bitset, cfg);
+
+        if !enable_quorum_check {
```

Comment:
If you don't negate the condition, you'll be able to remove the label putting the check inside the if

Reply:
Which `if`? Do you mean  to change it to
```
if enable_quorum_check && total < target_quorum {
```